prime-router/src/main/kotlin/azure/SendFunction.kt

```diff
@@ -50,24 +55,43 @@ class SendFunction(private val workflowEngine: WorkflowEngine = WorkflowEngine()
                     .filterIndexed { i, _ -> retryToken == null || retryToken.transports.find { it.index == i } != null }
                     .forEachIndexed { i, transport ->
                         val retryItems = retryToken?.transports?.find { it.index == i }?.items
+                        val sentReportId = UUID.randomUUID() // each sent report gets its own UUID
```

Comment:
This is a significant change.    The act of transmitting a file is now tracked as a parent-child relationship, just like any other.   Here, the parent is the blob stored in the Hub, the `inputReportId` to the `send` function.   And the child gets a new UUID here, the `sendReportId`, which will be used as the PK of a report_file entry representing the file as transmitted.   Note that this lets us track both failed and successful transmissions.